File: sample/src/main/java/com/applandeo/materialcalendarsampleapp/CalendarActivity.java
package com.applandeo.materialcalendarsampleapp;

import android.os.Bundle;
import android.widget.Button;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.applandeo.materialcalendarsampleapp.utils.DrawableUtils;
import com.applandeo.materialcalendarview.CalendarDay;
import com.applandeo.materialcalendarview.CalendarView;
import com.applandeo.materialcalendarview.exceptions.OutOfDateRangeException;
import com.applandeo.materialcalendarview.utils.DateUtils;

import java.util.ArrayList;
import java.util.Calendar;
import java.util.List;
import java.util.Random;

/**
 * Created by Mateusz Kornakiewicz on 26.05.2017.
 */

public class CalendarActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.calendar_activity);

        List<CalendarDay> events = new ArrayList<>();

        Calendar calendar = Calendar.getInstance();
        CalendarDay calendarDay = new CalendarDay(calendar);
        calendarDay.setImageDrawable(DrawableUtils.getCircleDrawableWithText(this, "M"));
        events.add(calendarDay);

        Calendar calendar1 = Calendar.getInstance();
        calendar1.add(Calendar.DAY_OF_MONTH, 10);
        CalendarDay calendarDay1 = new CalendarDay(calendar1);
        calendarDay1.setImageResource(R.drawable.sample_icon_2);
        events.add(calendarDay1);

        Calendar calendar2 = Calendar.getInstance();
        calendar2.add(Calendar.DAY_OF_MONTH, 10);
        CalendarDay calendarDay2 = new CalendarDay(calendar2);
        calendarDay2.setImageResource(R.drawable.sample_icon_3);
        events.add(calendarDay2);

        Calendar calendar3 = Calendar.getInstance();
        calendar3.add(Calendar.DAY_OF_MONTH, 7);
        CalendarDay calendarDay3 = new CalendarDay(calendar3);
        calendarDay3.setImageResource(R.drawable.sample_four_icons);
        events.add(calendarDay3);

        Calendar calendar4 = Calendar.getInstance();
        calendar4.add(Calendar.DAY_OF_MONTH, 13);
        CalendarDay calendarDay4 = new CalendarDay(calendar4);
        calendarDay4.setImageDrawable(DrawableUtils.getThreeDots(this));
        events.add(calendarDay4);

        CalendarView calendarView = (CalendarView) findViewById(R.id.calendarView);

        Calendar min = Calendar.getInstance();
        min.add(Calendar.MONTH, -2);

        Calendar max = Calendar.getInstance();
        max.add(Calendar.MONTH, 2);

        calendarView.setMinimumDate(min);
        calendarView.setMaximumDate(max);

        calendarView.setCalendarDays(events);

        calendarView.setDisabledDays(getDisabledDays());

        calendarView.setOnDayClickListener(eventDay ->
                Toast.makeText(getApplicationContext(),
                        eventDay.getCalendar().getTime().toString() + " "
                                + eventDay.isEnabled(),
                        Toast.LENGTH_SHORT).show());

        Button setDateButton = (Button) findViewById(R.id.setDateButton);
        setDateButton.setOnClickListener(v -> {
            try {
                Calendar randomCalendar = getRandomCalendar();
                String text = randomCalendar.getTime().toString();
                Toast.makeText(getApplicationContext(), text, Toast.LENGTH_LONG).show();
                calendarView.setDate(randomCalendar);
            } catch (OutOfDateRangeException exception) {
                exception.printStackTrace();

                Toast.makeText(getApplicationContext(),
                        "Date is out of range",
                        Toast.LENGTH_LONG).show();
            }
        });
    }

    private List<Calendar> getDisabledDays() {
        Calendar firstDisabled = DateUtils.getMidnightCalendar();
        firstDisabled.add(Calendar.DAY_OF_MONTH, 2);

        Calendar secondDisabled = DateUtils.getMidnightCalendar();
        secondDisabled.add(Calendar.DAY_OF_MONTH, 1);

        Calendar thirdDisabled = DateUtils.getMidnightCalendar();
        thirdDisabled.add(Calendar.DAY_OF_MONTH, 18);

        List<Calendar> calendars = new ArrayList<>();
        calendars.add(firstDisabled);
        calendars.add(secondDisabled);
        calendars.add(thirdDisabled);
        return calendars;
    }

    private Calendar getRandomCalendar() {
        Random random = new Random();

        Calendar calendar = Calendar.getInstance();
        calendar.add(Calendar.MONTH, random.nextInt(99));

        return calendar;
    }
}

File: sample/src/main/java/com/applandeo/materialcalendarsampleapp/CalendarActivity.kt
package com.applandeo.materialcalendarsampleapp

import android.graphics.Color
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import com.applandeo.materialcalendarsampleapp.utils.getCircleDrawableWithText
import com.applandeo.materialcalendarsampleapp.utils.getThreeDots
import com.applandeo.materialcalendarview.EventDay
import com.applandeo.materialcalendarview.exceptions.OutOfDateRangeException
import com.applandeo.materialcalendarview.listeners.OnDayClickListener
import com.applandeo.materialcalendarview.utils.DateUtils
import kotlinx.android.synthetic.main.calendar_activity.*
import java.util.*

/**
 * Created by Mateusz Kornakiewicz on 26.05.2017.
 */

class CalendarActivity : AppCompatActivity() {

    private val disabledDays: List<Calendar>
        get() {
            val firstDisabled = DateUtils.calendar.apply { add(Calendar.DAY_OF_MONTH, 2) }
            val secondDisabled = DateUtils.calendar.apply { add(Calendar.DAY_OF_MONTH, 1) }
            val thirdDisabled = DateUtils.calendar.apply { add(Calendar.DAY_OF_MONTH, 18) }

            return ArrayList<Calendar>().apply {
                add(firstDisabled)
                add(secondDisabled)
                add(thirdDisabled)
            }
        }

    private fun getRandomCalendar(): Calendar {
        val calendar = Calendar.getInstance()
        calendar.add(Calendar.MONTH, Random().nextInt(99))

        return calendar
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.calendar_activity)

        val events = ArrayList<EventDay>()

        val calendar = Calendar.getInstance()
        events.add(EventDay(calendar,
                this.getCircleDrawableWithText("M"),
                Color.parseColor("#228B22")))

        val calendar1 = Calendar.getInstance().apply { add(Calendar.DAY_OF_MONTH, 2) }
        val calendar2 = Calendar.getInstance().apply { add(Calendar.DAY_OF_MONTH, 5) }
        val calendar3 = Calendar.getInstance().apply { add(Calendar.DAY_OF_MONTH, 7) }
        val calendar4 = Calendar.getInstance().apply { add(Calendar.DAY_OF_MONTH, 13) }

        events.apply {
            add(EventDay(calendar1, R.drawable.sample_icon_2, Color.parseColor("#228B22")))
            add(EventDay(calendar2, R.drawable.sample_icon_3, Color.parseColor("#228B22")))
            add(EventDay(calendar3, R.drawable.sample_four_icons, Color.parseColor("#228B22")))
            add(EventDay(calendar4, (this@CalendarActivity).getThreeDots(), Color.parseColor("#228B22")))
        }

        val min = Calendar.getInstance().apply { add(Calendar.MONTH, -2) }
        val max = Calendar.getInstance().apply { add(Calendar.MONTH, 2) }

        calendarView.apply {
            setMinimumDate(min)
            setMaximumDate(max)
            setEvents(events)
            setDisabledDays(disabledDays)
            setOnDayClickListener(object : OnDayClickListener {
                override fun onDayClick(eventDay: EventDay) {
                    Toast.makeText(applicationContext, getTitle(eventDay), Toast.LENGTH_SHORT).show()
                }
            })
        }

        val setDateButton = findViewById<View>(R.id.setDateButton) as Button
        setDateButton.setOnClickListener {
            try {
                val randomCalendar = getRandomCalendar()
                val text = randomCalendar.time.toString()
                Toast.makeText(applicationContext, text, Toast.LENGTH_LONG).show()
                calendarView.setDate(randomCalendar)
            } catch (exception: OutOfDateRangeException) {
                exception.printStackTrace()

                Toast.makeText(applicationContext, "Date is out of range", Toast.LENGTH_LONG).show()
            }
        }
    }

    private fun getTitle(eventDay: EventDay) =
            "${eventDay.calendar?.time.toString()} ${eventDay.isEnabled}"
}
